### PR TITLE
Replace signing with signature before validation

### PR DESCRIPTION
**Pull request by OrfeasLitos**

### CODE REVIEW

**Review by jonasnick — APPROVED** [review]

**Review by real-or-random — APPROVED** [review]
ACK 

Actually we renamed validation to verification in bip-schnorr, so we may want to do this here was well for consistency.

**Comment by jonasnick** [comment]
> Actually we renamed validation to verification in bip-schnorr, so we may want to do this here was well for consistency.

Not necessarily because taproot signature validation does much more than just signature verification.

File: bip-tapscript.mediawiki [hunk 1]
```diff
@@ -89,7 +89,7 @@ The following rules apply to <code>OP_CHECKSIG</code>, <code>OP_CHECKSIGVERIFY</
...
+** If the signature is not the empty vector, the signature is validated according to the bip-taproot signature validation rules against the public key and the tapscript transaction digest (to be defined hereinafter) as message. Validation failure MUST cause the script to fail and terminate immediately.
 * If the public key size is not zero and not 32 bytes, the public key is of an ''unknown public key type''<ref>'''Unknown public key types''' allow adding new signature validation rules through softforks. A softfork could add actual signature validation which either passes or makes the script fail and terminate immediately. This way, new <code>SIGHASH</code> modes can be added, as well as [https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2018-December/016549.html NOINPUT-tagged public keys] and a public key constant which is replaced by the taproot internal key for signature validation.</ref> and no actual signature verification is applied. During script execution of signature opcodes they behave exactly as known public key types except that signature validation is considered to be successful.
```
Comment (real-or-random) [inline bip-tapscript.mediawiki]: Here in this point I'm not sure if verification and validation are used according to this convention.

**Review by real-or-random — COMMENTED** [review]
https://github.com/sipa/bips/pull/162/files#diff-506cf22853a59a8c57550c1f1d9cedfbL131

Here we say script verification but this should be validation.

**Comment by real-or-random** [comment]
Also here in the first sentence of this paragraph I'm not sure:

https://github.com/sipa/bips/blob/bip-schnorr/bip-tapscript.mediawiki#transaction-digest

To be honest, I can't even parse the first clause of that sentence.

**Comment by OrfeasLitos** [comment]
> Also here in the first sentence of this paragraph I'm not sure:
> 
> https://github.com/sipa/bips/blob/bip-schnorr/bip-tapscript.mediawiki#transaction-digest
> 
> To be honest, I can't even parse the first clause of that sentence.

Could be rephrased to "As the message used in the signature verification of signature opcodes"

**Pull request merged**